diff --git a/configure b/configure
index 4731375..04bcf67 100755
--- a/configure
+++ b/configure
@@ -3258,9 +3258,6 @@ host_os=$*
 IFS=$ac_save_IFS
 case $host_os in *\ *) host_os=`echo "$host_os" | sed 's/ /-/g'`;; esac
 
-
-
-
 opt_bash_malloc=yes
 opt_afs=no
 opt_curses=no
@@ -3281,6 +3278,7 @@ m68k-sysv)	opt_bash_malloc=no ;;	# fixes file descriptor leak in closedir
 *-beos*)	opt_bash_malloc=no ;;	# they say it's suitable
 # These need additional investigation
 sparc-linux*)	opt_bash_malloc=no ;;	# sparc running linux; requires ELF
+i370-*)		opt_bash_malloc=no ;;	# z/OS machines
 *-aix*)		opt_bash_malloc=no ;;	# AIX machines
 *-cygwin*)	opt_bash_malloc=no ;;	# Cygnus's CYGWIN environment
 # These lack a working sbrk(2)
@@ -21600,13 +21598,16 @@ then :
   printf %s "(cached) " >&6
 else $as_nop
   bash_cv_dev_fd=""
-if test -d /dev/fd  && (exec test -r /dev/fd/0 < /dev/null) ; then
+# Some z/OS systems may not have a /dev/fd directory
+if test -d /dev/fd  && (exec test -r /dev/fd/0 < /dev/null) && test "$host_os" != "openedition"; then
 # check for systems like FreeBSD 5 that only provide /dev/fd/[012]
-   if (exec test -r /dev/fd/3 3</dev/null) ; then
+   if (exec test -r /dev/fd/3 3</dev/null); then
      bash_cv_dev_fd=standard
    else
      bash_cv_dev_fd=absent
    fi
+elif test "$host_os" = "openedition" && (exec test -r /dev/fd0 < /dev/null); then
+  bash_cv_dev_fd=standard
 fi
 if test -z "$bash_cv_dev_fd" ; then
   if test -d /proc/self/fd && (exec test -r /proc/self/fd/0 < /dev/null) ; then
@@ -21623,7 +21624,11 @@ printf "%s\n" "$bash_cv_dev_fd" >&6; }
 if test $bash_cv_dev_fd = "standard"; then
   printf "%s\n" "#define HAVE_DEV_FD 1" >>confdefs.h
 
+if test "$host_os" = "openedition" && (exec test -r /dev/fd0 < /dev/null); then
+  printf "%s\n" "#define DEV_FD_PREFIX \"/dev/fd\"" >>confdefs.h
+else
   printf "%s\n" "#define DEV_FD_PREFIX \"/dev/fd/\"" >>confdefs.h
+fi
 
 elif test $bash_cv_dev_fd = "whacky"; then
   printf "%s\n" "#define HAVE_DEV_FD 1" >>confdefs.h
