diff --git a/lib/glob/smatch.c b/lib/glob/smatch.c
index 379c2d2..acb436f 100644
--- a/lib/glob/smatch.c
+++ b/lib/glob/smatch.c
@@ -141,7 +141,7 @@ rangecmp (c1, c2, forcecoll)
 #  define rangecmp(c1, c2, f)	((int)(c1) - (int)(c2))
 #endif /* !HAVE_STRCOLL */
 
-#if defined (HAVE_STRCOLL)
+#if defined (HAVE_STRCOLL) && !defined(__MVS__)
 /* Returns 1 if chars C and EQUIV collate equally in the current locale. */
 static int
 collequiv (c, equiv)
@@ -157,7 +157,7 @@ collequiv (c, equiv)
 #endif
   
 }
-#else
+#elif !defined(__MVS__)
 #  define collequiv(c, equiv)	((c) == (equiv))
 #endif
 
diff --git a/lib/intl/gettextP.h b/lib/intl/gettextP.h
index 5c137e5..9a9d840 100644
--- a/lib/intl/gettextP.h
+++ b/lib/intl/gettextP.h
@@ -77,6 +77,9 @@ SWAP (i)
 }
 #endif
 
+#ifdef __MVS__
+#undef __locale
+#endif
 
 /* In-memory representation of system dependent string.  */
 struct sysdep_string_desc
@@ -169,6 +172,10 @@ struct loaded_l10nfile *_nl_find_domain PARAMS ((const char *__dirname,
 						 const char *__domainname,
 					      struct binding *__domainbinding))
      internal_function;
+
+#ifdef __MVS__
+#define __locale 1
+#endif
 void _nl_load_domain PARAMS ((struct loaded_l10nfile *__domain,
 			      struct binding *__domainbinding))
      internal_function;
diff --git a/print_cmd.c b/print_cmd.c
index eef9bb6..a6a15b8 100644
--- a/print_cmd.c
+++ b/print_cmd.c
@@ -59,8 +59,8 @@ static int indentation_amount = 4;
 #if defined (PREFER_STDARG)
 typedef void PFUNC PARAMS((const char *, ...));
 
-static void cprintf PARAMS((const char *, ...))  __attribute__((__format__ (printf, 1, 2)));
-static void xprintf PARAMS((const char *, ...))  __attribute__((__format__ (printf, 1, 2)));
+static void cprintf (const char *control, ...);
+static void xprintf (const char *, ...);
 #else
 #define PFUNC VFunction
 static void cprintf ();
diff --git a/y.tab.c b/y.tab.c
index 32b4c7c..caa4f1b 100644
--- a/y.tab.c
+++ b/y.tab.c
@@ -131,7 +131,7 @@ typedef void *alias_t;
 #endif /* ALIAS */
 
 #if defined (PROMPT_STRING_DECODE)
-#  ifndef _MINIX
+#  if !defined(_MINIX) && !defined(__MVS__)
 #    include <sys/param.h>
 #  endif
 #  include <time.h>
