diff --git a/tests/intl.right b/tests/intl.right
index 7da9919..5e1fb67 100644
--- a/tests/intl.right
+++ b/tests/intl.right
@@ -51,7 +51,7 @@ Passed all 1378 Unicode tests
 0000000 101 040 302 243 040 305 222 012
 0000010
 ./unicode3.sub: line 5: $'5\247@3\231+\306S8\237\242\352\263': command not found
-./unicode3.sub: line 7: cd: $'5\247@3\231+\306S8\237\242\352\263': No such file or directory
+./unicode3.sub: line 7: cd: $'5\247@3\231+\306S8\237\242\352\263': EDC5129I No such file or directory.
 $'5\247@3\231+\306S8\237\242\352\263'
 + : $'5\247@3\231+\306S8\237\242\352\263'
 + set +x
