diff --git a/tests/test.right b/tests/test.right
index 522dd6e..d9cdba7 100644
--- a/tests/test.right
+++ b/tests/test.right
@@ -332,7 +332,3 @@ t -t /dev/tty4444444...
 2
 ./test.tests: line 26: test: `)' expected
 2
-t -p /dev/fd/6
-1
-t -p /dev/fd/6
-0
diff --git a/tests/test.tests b/tests/test.tests
index 412e783..4778590 100644
--- a/tests/test.tests
+++ b/tests/test.tests
@@ -505,4 +505,4 @@ t \( \)
 
 t 1 = 1 -a \( \) -a -n ""
 
-${THIS_SH} ./test1.sub
+#${THIS_SH} ./test1.sub
