diff --git a/tests/trap.right b/tests/trap.right
index 40c51dd..fed96b1 100644
--- a/tests/trap.right
+++ b/tests/trap.right
@@ -2,18 +2,20 @@ subshell exit
 trap -- 'echo exiting' EXIT
 trap -- 'echo aborting' SIGHUP
 trap -- 'echo aborting' SIGINT
-trap -- 'echo aborting' SIGQUIT
 trap -- 'echo aborting' SIGABRT
 trap -- 'echo aborting' SIGTERM
+trap -- 'echo aborting' SIGQUIT
+trap -- '' SIGIOERR
 [33] debug
 debug line
 [35] debug
 trap -- 'echo exiting' EXIT
 trap -- 'echo aborting' SIGHUP
 trap -- 'echo aborting' SIGINT
-trap -- 'echo aborting' SIGQUIT
 trap -- 'echo aborting' SIGABRT
 trap -- 'echo aborting' SIGTERM
+trap -- 'echo aborting' SIGQUIT
+trap -- '' SIGIOERR
 trap -- 'echo [$LINENO] debug' DEBUG
 [37] debug
 func[29] funcdebug
@@ -22,9 +24,10 @@ funcdebug line
 trap -- 'echo exiting' EXIT
 trap -- 'echo aborting' SIGHUP
 trap -- 'echo aborting' SIGINT
-trap -- 'echo aborting' SIGQUIT
 trap -- 'echo aborting' SIGABRT
 trap -- 'echo aborting' SIGTERM
+trap -- 'echo aborting' SIGQUIT
+trap -- '' SIGIOERR
 trap -- 'echo ${FUNCNAME:-$0}[$LINENO] funcdebug' DEBUG
 ./trap.tests[41] funcdebug
 ./trap.tests[46] debug
@@ -37,21 +40,24 @@ func2debug line
 trap -- 'echo exiting' EXIT
 trap -- 'echo aborting' SIGHUP
 trap -- 'echo aborting' SIGINT
-trap -- 'echo aborting' SIGQUIT
 trap -- 'echo aborting' SIGABRT
 trap -- 'echo aborting' SIGTERM
+trap -- 'echo aborting' SIGQUIT
+trap -- '' SIGIOERR
 trap -- '' DEBUG
 trap -- 'echo exiting' EXIT
 trap -- 'echo aborting' SIGHUP
 trap -- 'echo aborting' SIGINT
-trap -- 'echo aborting' SIGQUIT
 trap -- 'echo aborting' SIGABRT
 trap -- 'echo aborting' SIGTERM
+trap -- 'echo aborting' SIGQUIT
+trap -- '' SIGIOERR
 trap -- 'echo exiting' EXIT
 trap -- '' SIGINT
-trap -- 'echo aborting' SIGQUIT
 trap -- 'echo aborting' SIGABRT
 trap -- 'echo aborting' SIGTERM
+trap -- 'echo aborting' SIGQUIT
+trap -- '' SIGIOERR
 0
 trap -- '' SIGUSR2
 ERRTRAP
@@ -133,6 +139,7 @@ trap: usage: trap [-Plp] [[action] signal_spec ...]
 trap -- 'echo exiting' EXIT
 trap -- 'echo aborting' SIGABRT
 trap -- 'echo caught a child death' SIGCHLD
+trap -- '' SIGIOERR
 trap: usage: trap [-Plp] [[action] signal_spec ...]
 echo exiting
 exiting
