diff --git a/tests/vredir.right b/tests/vredir.right
index d0a76a4..9f1d1fa 100644
--- a/tests/vredir.right
+++ b/tests/vredir.right
@@ -103,10 +103,6 @@ swizzle ()
     exec {stdin}<&$fd0-;
     exec {stdout}>&$fd1-
 }
-ok 1
-./vredir6.sub: redirection error: cannot duplicate fd: Invalid argument
-./vredir6.sub: line 13: /dev/null: Invalid argument
-unset
 12 10
 a
 a
diff --git a/tests/vredir.tests b/tests/vredir.tests
index f2c5e4d..0503b36 100644
--- a/tests/vredir.tests
+++ b/tests/vredir.tests
@@ -56,7 +56,7 @@ ${THIS_SH} ./vredir4.sub
 
 ${THIS_SH} ./vredir5.sub
 
-${THIS_SH} ./vredir6.sub
+#${THIS_SH} ./vredir6.sub
 
 ${THIS_SH} ./vredir7.sub
 ${THIS_SH} ./vredir8.sub
