diff --git a/aclocal.m4 b/aclocal.m4
index 2a86c70c..5473e4f5 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -1573,13 +1573,15 @@ AC_DEFUN(BASH_CHECK_DEV_FD,
 [AC_MSG_CHECKING(whether /dev/fd is available)
 AC_CACHE_VAL(bash_cv_dev_fd,
 [bash_cv_dev_fd=""
-if test -d /dev/fd  && (exec test -r /dev/fd/0 < /dev/null) ; then
+if test -d /dev/fd  && (exec test -r /dev/fd/0 < /dev/null) && test "$host_os" != "openedition"; then
 # check for systems like FreeBSD 5 that only provide /dev/fd/[012]
    if (exec test -r /dev/fd/3 3</dev/null) ; then
      bash_cv_dev_fd=standard
    else
      bash_cv_dev_fd=absent
    fi
+elif test "$host_os" = "openedition" && (exec test -r /dev/fd0 < /dev/null); then
+  bash_cv_dev_fd=standard
 fi
 if test -z "$bash_cv_dev_fd" ; then 
   if test -d /proc/self/fd && (exec test -r /proc/self/fd/0 < /dev/null) ; then
@@ -1592,7 +1594,11 @@ fi
 AC_MSG_RESULT($bash_cv_dev_fd)
 if test $bash_cv_dev_fd = "standard"; then
   AC_DEFINE(HAVE_DEV_FD)
+if test "$host_os" = "openedition"; then
+  AC_DEFINE(DEV_FD_PREFIX, "/dev/fd")
+else
   AC_DEFINE(DEV_FD_PREFIX, "/dev/fd/")
+fi
 elif test $bash_cv_dev_fd = "whacky"; then
   AC_DEFINE(HAVE_DEV_FD)
   AC_DEFINE(DEV_FD_PREFIX, "/proc/self/fd/")
