diff --git a/tests/builtins.right b/tests/builtins.right
index 4f51d43..0a91d67 100644
--- a/tests/builtins.right
+++ b/tests/builtins.right
@@ -106,9 +106,9 @@ m n o p
 /tmp/bash-dir-a
 /tmp/bash-dir-a
 /tmp/bash-dir-a
-./source5.sub: line 23: /tmp/source-notthere: No such file or directory
+./source5.sub: line 23: /tmp/source-notthere: EDC5129I No such file or directory.
 after bad source 1
-./source5.sub: line 30: /tmp/source-notthere: No such file or directory
+./source5.sub: line 30: /tmp/source-notthere: EDC5129I No such file or directory.
 one - OK
 0
 0
@@ -242,9 +242,9 @@ f2 ()
 funcs unset:
 one-two
 two-two
-bash: line 1: notthere: No such file or directory
+bash: line 1: notthere: EDC5129I No such file or directory.
 one
-bash: line 1: notthere: No such file or directory
+bash: line 1: notthere: EDC5129I No such file or directory.
 two
 bash: line 1: .: notthere: file not found
 one
@@ -264,9 +264,9 @@ four
 ./builtins7.sub: line 19: : command not found
 127
 0
-./builtins7.sub: line 27: notthere: No such file or directory
+./builtins7.sub: line 27: notthere: EDC5129I No such file or directory.
 after 1
-./builtins7.sub: line 28: notthere: No such file or directory
+./builtins7.sub: line 28: notthere: EDC5129I No such file or directory.
 after 2
 type is a shell builtin
 + command -v type
