diff --git a/tests/coproc.tests b/tests/coproc.tests
index a735932..64139cf 100644
--- a/tests/coproc.tests
+++ b/tests/coproc.tests
@@ -38,6 +38,7 @@ esac
 echo ${REFLECT[@]}
 
 echo flop >&${REFLECT[1]}
+
 read LINE <&${REFLECT[0]}
 
 echo $LINE
@@ -65,7 +66,7 @@ read <&${COPROC[0]}
 echo $REPLY
 echo ${COPROC[@]}
 
-cat /etc/passwd | grep root | awk -F: '{print $1;}' | sed 1q
+cat /etc/hosts | grep localhost | awk '/127.0.0.1/ {print root;}' | sed 1q
 
 exec 4<&${COPROC[0]}-
 exec >&${COPROC[1]}-
