diff --git a/configure b/configure
index a28a3fc..39ebb65 100755
--- a/configure
+++ b/configure
@@ -3326,6 +3326,7 @@ m68k-sysv)	opt_bash_malloc=no ;;	# fixes file descriptor leak in closedir
 *-beos*)	opt_bash_malloc=no ;;	# they say it's suitable
 # These need additional investigation
 sparc-linux*)	opt_bash_malloc=no ;;	# sparc running linux; requires ELF
+i370-*)         opt_bash_malloc=no ;;   # z/OS machines
 *-aix*)		opt_bash_malloc=no ;;	# AIX machines
 *-cygwin*|msys*) opt_bash_malloc=no ;;	# Cygnus's CYGWIN environment
 # These lack a working sbrk(2)
@@ -23012,15 +23013,16 @@ then :
   printf %s "(cached) " >&6
 else case e in #(
   e) bash_cv_dev_fd=""
-if test -d /dev/fd && (exec test -r /dev/fd/0 < /dev/null) ; then
+# Some z/OS systems may not have a /dev/fd directory
+if test -d /dev/fd  && (exec test -r /dev/fd/0 < /dev/null) && test "$host_os" != "openedition"; then
 # check for systems like FreeBSD 5 that only provide /dev/fd/[012]
-   if (exec test -r /dev/fd/3 3</dev/null) ; then
+   if (exec test -r /dev/fd/3 3</dev/null); then
      bash_cv_dev_fd=standard
    else
      bash_cv_dev_fd=absent
    fi
 elif test "$host_os" = "openedition" && (exec test -r /dev/fd0 < /dev/null); then
-  bash_cv_dev_fd=nodir		# /dev/fdN via character device
+  bash_cv_dev_fd=standard
 fi
 if test -z "$bash_cv_dev_fd" ; then
   if test -d /proc/self/fd && (exec test -r /proc/self/fd/0 < /dev/null) ; then
@@ -23038,7 +23040,11 @@ printf "%s\n" "$bash_cv_dev_fd" >&6; }
 if test $bash_cv_dev_fd = "standard"; then
   printf "%s\n" "#define HAVE_DEV_FD 1" >>confdefs.h
 
+if test "$host_os" = "openedition" && (exec test -r /dev/fd0 < /dev/null); then
+  printf "%s\n" "#define DEV_FD_PREFIX \"/dev/fd\"" >>confdefs.h
+else
   printf "%s\n" "#define DEV_FD_PREFIX \"/dev/fd/\"" >>confdefs.h
+fi
 
 elif test $bash_cv_dev_fd = "whacky"; then
   printf "%s\n" "#define HAVE_DEV_FD 1" >>confdefs.h
@@ -23067,6 +23073,9 @@ else case e in #(
 esac
 fi
 
+# z/OS workaround since /dev/stdin is broken in 3.1
+   bash_cv_dev_stdin=absent
+
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $bash_cv_dev_stdin" >&5
 printf "%s\n" "$bash_cv_dev_stdin" >&6; }
 if test $bash_cv_dev_stdin = "present"; then
