diff --git a/lib/intl/gettextP.h b/lib/intl/gettextP.h
index fc21582..8e262d9 100644
--- a/lib/intl/gettextP.h
+++ b/lib/intl/gettextP.h
@@ -119,6 +119,9 @@ SWAP (nls_uint32 i)
 }
 #endif
 
+#ifdef __MVS__
+#undef __locale
+#endif
 
 /* In-memory representation of system dependent string.  */
 struct sysdep_string_desc
@@ -262,6 +265,10 @@ struct loaded_l10nfile *_nl_find_domain (const char *__dirname,
 					 const char *__domainname,
 					 struct binding *__domainbinding)
      internal_function;
+
+#ifdef __MVS__
+#define __locale 1
+#endif
 void _nl_load_domain (struct loaded_l10nfile *__domain,
 		      struct binding *__domainbinding)
      internal_function;
