diff --git a/MANIFEST b/MANIFEST
index 985d179..2a1cd25 100644
--- a/MANIFEST
+++ b/MANIFEST
@@ -1455,6 +1455,10 @@ tests/quotearray2.sub	f
 tests/quotearray3.sub	f
 tests/quotearray4.sub	f
 tests/quotearray5.sub	f
+tests/encodefilevars.tests	f
+tests/encodefilevars.right	f
+tests/rbash.tests	f
+tests/rbash.right	f
 tests/read.tests	f
 tests/read.right	f
 tests/read1.sub		f
@@ -1565,6 +1569,8 @@ tests/run-printf	f
 tests/run-procsub	f
 tests/run-quote		f
 tests/run-quotearray	f
+tests/run-encodefilevars	f
+tests/run-rbash	f
 tests/run-read		f
 tests/run-redir		f
 tests/run-rhs-exp	f
