diff --git a/tests/builtins.right b/tests/builtins.right
index d708c18..52fad9c 100644
--- a/tests/builtins.right
+++ b/tests/builtins.right
@@ -110,9 +110,9 @@ m n o p
 /tmp/bash-dir-a
 /tmp/bash-dir-a
 /
-./source5.sub: line 23: /tmp/source-notthere: No such file or directory
+./source5.sub: line 23: /tmp/source-notthere: EDC5129I No such file or directory
 after bad source 1
-./source5.sub: line 30: /tmp/source-notthere: No such file or directory
+./source5.sub: line 30: /tmp/source-notthere: EDC5129I No such file or directory
 one - OK
 0
 0
@@ -139,8 +139,8 @@ abc
 def
 ghi
 ok
-./source8.sub: line 40: improbable-filename: No such file or directory
-./source8.sub: line 41: improbable-filename: No such file or directory
+./source8.sub: line 40: improbable-filename: EDC5129I No such file or directory
+./source8.sub: line 41: improbable-filename: EDC5129I No such file or directory
 an improbable filename
 an improbable filename
 an improbable filename
@@ -261,9 +261,9 @@ f2 ()
 funcs unset:
 one-two
 two-two
-bash: line 1: notthere: No such file or directory
+bash: line 1: notthere: EDC5129I No such file or directory.
 one
-bash: line 1: notthere: No such file or directory
+bash: line 1: notthere: EDC5129I No such file or directory.
 two
 bash: line 1: .: notthere: file not found
 one
@@ -283,9 +283,9 @@ four
 ./builtins7.sub: line 19: : command not found
 127
 0
-./builtins7.sub: line 27: notthere: No such file or directory
+./builtins7.sub: line 27: notthere: EDC5129I No such file or directory.
 after 1
-./builtins7.sub: line 28: notthere: No such file or directory
+./builtins7.sub: line 28: notthere: EDC5129I No such file or directory.
 after 2
 type is a shell builtin
 + command -v type
@@ -325,13 +325,13 @@ builtin hash -p /nosuchdir/nosuchfile cat
 ./builtins9.sub: line 32: hash: notthere: not found
 ./builtins9.sub: line 33: hash: notthere: not found
 1
-./builtins9.sub: line 40: /nosuchdir/nosuchfile: No such file or directory
+./builtins9.sub: line 40: /nosuchdir/nosuchfile: EDC5129I No such file or directory
 127
 /nosuchdir/nosuchfile
 0
 0
 found
-./builtins9.sub: line 52: hash: /: Is a directory
+./builtins9.sub: line 52: hash: /: EDC5123I Is a directory
 builtin hash -p /nosuchfile cat
 ./builtins10.sub: line 17: help: -x: invalid option
 help: usage: help [-dms] [pattern ...]
