diff --git a/tests/dstack.right b/tests/dstack.right
index 8f9d38c..82d1809 100644
--- a/tests/dstack.right
+++ b/tests/dstack.right
@@ -1,4 +1,4 @@
-./dstack.tests: line 19: pushd: /tmp/xxx-notthere: No such file or directory
+./dstack.tests: line 19: pushd: /tmp/xxx-notthere: EDC5129I No such file or directory.
 ./dstack.tests: line 22: pushd: no other directory
 ./dstack.tests: line 23: popd: directory stack empty
 ./dstack.tests: line 26: pushd: -m: invalid number
