diff --git a/tests/errors.right b/tests/errors.right
index f46ebb1..5e78099 100644
--- a/tests/errors.right
+++ b/tests/errors.right
@@ -71,10 +71,10 @@ comsub: -c: line 1: `: $( for z in 1 2 3; do )'
 comsub: -c: line 1: syntax error near unexpected token `done' while looking for matching `)'
 comsub: -c: line 1: `: $( for z in 1 2 3; done )'
 ./errors.tests: line 221: cd: HOME not set
-./errors.tests: line 222: cd: /tmp/xyz.bash: No such file or directory
+./errors.tests: line 222: cd: /tmp/xyz.bash: EDC5129I No such file or directory
 ./errors.tests: line 224: cd: OLDPWD not set
-./errors.tests: line 225: cd: /bin/sh: Not a directory
-./errors.tests: line 227: cd: /tmp/cd-notthere: No such file or directory
+./errors.tests: line 225: cd: /bin/sh: EDC5135I Not a directory
+./errors.tests: line 227: cd: /tmp/cd-notthere: EDC5129I No such file or directory
 ./errors.tests: line 229: cd: too many arguments
 bash: line 1: PWD: readonly variable
 1
@@ -151,9 +151,9 @@ set: usage: set [-abefhkmnptuvxBCEHPT] [-o option-name] [--] [-] [arg ...]
 after f
 ./errors2.sub: line 3: ${$NO_SUCH_VAR}: bad substitution
 1
-./errors3.sub: line 5: no_such_file: No such file or directory
+./errors3.sub: line 5: no_such_file: EDC5129I No such file or directory.
 TEST
-./errors3.sub: line 7: no_such_file: No such file or directory
+./errors3.sub: line 7: no_such_file: EDC5129I No such file or directory.
 1
 2
 ./errors4.sub: line 20: var: readonly variable
