diff --git a/tests/exec.right b/tests/exec.right
index 0116f20..32c7579 100644
--- a/tests/exec.right
+++ b/tests/exec.right
@@ -8,13 +8,13 @@ after exec1.sub: one two three
 ./execscript: line 37: notthere: command not found
 127
 we would do something here with notthere
-/tmp/bash: notthere: No such file or directory
+/tmp/bash: notthere: EDC5129I No such file or directory
 127
 /bin/sh: /bin/sh: cannot execute binary file
 126
-./execscript: line 64: /: Is a directory
+./execscript: line 64: /: EDC5123I Is a directory
 126
-/: /: Is a directory
+/: /: EDC5123I Is a directory
 126
 bash: line 1: exec: .: cannot execute: Is a directory
 posix-bash: line 1: exec: .: cannot execute: Is a directory
@@ -29,7 +29,7 @@ trap -- 'echo EXIT' EXIT
 trap -- '' SIGTERM
 trap -- 'echo USR1' SIGUSR1
 USR1
-./exec3.sub: line 38: /tmp/bash-notthere: No such file or directory
+./exec3.sub: line 38: /tmp/bash-notthere: EDC5129I No such file or directory
 ./exec3.sub: ENOENT: after failed exec: 127
 ./exec3.sub: line 43: exec: bash-notthere: not found
 trap -- 'echo EXIT' EXIT
