diff --git a/tests/new-exp.right b/tests/new-exp.right
index e3dc40e..597143a 100644
--- a/tests/new-exp.right
+++ b/tests/new-exp.right
@@ -170,7 +170,7 @@ This-string-has-multiple-lines.
 this is a test of proc subst
 this is test 2
 a
-./new-exp2.sub: line 55: /tmp/redir-notthere: No such file or directory
+./new-exp2.sub: line 55: /tmp/redir-notthere: EDC5129I No such file or directory.
 1
 
 ./new-exp2.sub: line 62: 1111111111111111111111: command not found
