diff --git a/tests/redir.right b/tests/redir.right
index c5d3779..f40ed57 100644
--- a/tests/redir.right
+++ b/tests/redir.right
@@ -48,7 +48,7 @@ kl
 ab
 cd
 cd
-./redir.tests: line 184: redir1.*: No such file or directory
+./redir.tests: line 184: redir1.*: EDC5129I No such file or directory
 #   This program is free software: you can redistribute it and/or modify
 #   it under the terms of the GNU General Public License as published by
 #   the Free Software Foundation, either version 3 of the License, or
@@ -165,20 +165,20 @@ after: 42
 a+=3
 foo
 foo
-./redir11.sub: line 77: 42: No such file or directory
+./redir11.sub: line 77: 42: EDC5129I No such file or directory
 42
-./redir12.sub: line 27: unwritable-file: Permission denied
+./redir12.sub: line 27: unwritable-file: EDC5111I Permission denied
 1 x =
 ./redir12.sub: line 31: unwritable-file: cannot overwrite existing file
 1 x =
-./redir12.sub: line 34: unwritable-file: Permission denied
+./redir12.sub: line 34: unwritable-file: EDC5111I Permission denied
 1 x =
-./redir12.sub: line 38: unreadable-file: Permission denied
-./redir12.sub: line 44: unwritable-file: Permission denied
+./redir12.sub: line 38: unreadable-file: EDC5111I Permission denied
+./redir12.sub: line 44: unwritable-file: EDC5111I Permission denied
 got error ERR
 after ERR trap: 1
-./redir12.sub: line 48: unwritable-file: Permission denied
-./redir12.sub: line 53: unreadable-file: Permission denied
+./redir12.sub: line 48: unwritable-file: EDC5111I Permission denied
+./redir12.sub: line 53: unreadable-file: EDC5111I Permission denied
 got error ERR
-./redir12.sub: line 56: unreadable-file: Permission denied
+./redir12.sub: line 56: unreadable-file: EDC5111I Permission denied
 /tmp
