diff --git a/tests/vredir.right b/tests/vredir.right
index d0a76a4..db18dd7 100644
--- a/tests/vredir.right
+++ b/tests/vredir.right
@@ -91,7 +91,7 @@ swizzle ()
 }
 10 11
 10 11
-./vredir4.sub: line 51: ${output}: Bad file descriptor
+./vredir4.sub: line 51: ${output}: EDC5113I Bad file descriptor
 12 10
 a
 a
@@ -119,5 +119,5 @@ swizzle ()
     exec {stdout}>&${fd[1]}-
 }
 redir 2
-./vredir8.sub: line 33: $fd: Bad file descriptor
-./vredir8.sub: line 38: $fd: Bad file descriptor
+./vredir8.sub: line 33: $fd: EDC5113I Bad file descriptor
+./vredir8.sub: line 38: $fd: EDC5113I Bad file descriptor
