diff --git a/tests/vredir.tests b/tests/vredir.tests
index f2c5e4d..af489ac 100644
--- a/tests/vredir.tests
+++ b/tests/vredir.tests
@@ -37,9 +37,9 @@ readonly v=42
 bar
 exec {v}>&1
 
-echo foo 1 2>&1 >&$v | { grep -q '\$v: Bad' || echo 'bad foo 1'; }
-echo foo 2 2>&1 >&$v | { grep -q '\$v: Bad' || echo 'bad foo 2'; }
-echo foo 3 2>&1 >&$v | { grep -q '\$v: Bad' || echo 'bad foo 3'; }
+echo foo 1 2>&1 >&$v | { grep -q '\$v: EDC5113I Bad' || echo 'bad foo 1'; }
+echo foo 2 2>&1 >&$v | { grep -q '\$v: EDC5113I Bad' || echo 'bad foo 2'; }
+echo foo 3 2>&1 >&$v | { grep -q '\$v: EDC5113I Bad' || echo 'bad foo 3'; }
 
 cat $TMPFILE
 rm -f $TMPFILE
